### PR TITLE
Youssef branch

### PR DESCRIPTION
**Pull request by srondss**

### CODE REVIEW

File: Lib/libft/ftp_strjoin.c [hunk 1]
```diff
@@ -33,7 +34,12 @@ char	*ftp_strjoin(char *s1, char *s2)
 		s2_len = ft_strlen(s2);
 	str_join = malloc(sizeof(char) * (s1_len + s2_len + 1));
 	if (str_join == NULL)
-		return (free(s1), NULL);
+	{
+		printf("entering here?\n");
```
Comment (srondss) [inline Lib/libft/ftp_strjoin.c:38]: need to remove this printf

File: src/main.c [hunk 2]
```diff
@@ -26,32 +28,33 @@ int	main(int argc, char **argv, char **envp)
...
 	init_tools(tools, &tokens_head, &cmds_head);
 	init_tools_env(tools->env_list, envp);
-	g_exit_status = 0;
-
+	g_exit_status = 0; // is this needed?
 	while (tools->loop)
 	{
 		printf("--------NEW COMMAND---------------\n");
 		string = readline("Minishell: ");
+		if (!string)
```
Comment (Naji-k) [inline src/main.c:46]: segfault, because of `free(og_string)`,

File: src/main.c [hunk 3]
```diff
@@ -26,32 +28,33 @@ int	main(int argc, char **argv, char **envp)
...
+			return (malloc_error(NULL), free_all_exit(tools), g_exit_status);
+		add_history(string);
 		tools->og_string = ft_strdup(string);
 		if (!tools->og_string)
-			free_all_exit(tools);
-		add_history(string);
+		{
+			malloc_error(string);
```
Comment (Naji-k) [inline src/main.c:52]: `return (malloc_error(NULL), free_all_exit(tools), g_exit_status);` otherwise segfault

**Review by Naji-k — COMMENTED** [review]
SEGFAULT, when `ctl+d`, after any command
in mini_exit should `free(og_string)` or it will leaks,

**Pull request merged**